### PR TITLE
Allow precache to clear in background

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Allow prewarm cache to clear in background on prewarm continuation; only forcing at start of next block if not cleared

#### What types of changes does your code introduce?

- [x] Optimization

#### Requires testing

- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Consensus/Processing/BlockCachePreWarmer.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ public Task PreWarmCaches(Block suggestedBlock, Hash256? parentStateRoot, Access
         {
             if (targetWorldState.ClearCache())
             {
-                if (_logger.IsWarn) _logger.Warn("Caches are not empty. Clearing them.");
+                if (_logger.IsDebug) _logger.Debug("Caches are not empty. Clearing them.");
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Consensus/Processing/BlockCachePreWarmer.cs:34]: This generally should not happen in normal execution, why lower to Debug?

File: src/Nethermind/Nethermind.Core/Collections/CollectionExtensions.cs [hunk 2]
```diff
@@ -40,6 +40,12 @@ public static bool NoResizeClear<TKey, TValue>(this ConcurrentDictionary<TKey, T
 
             using var handle = dictionary.AcquireLock();
 
+            // Recheck under lock
+            if (dictionary is null || dictionary.IsEmpty)
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Core/Collections/CollectionExtensions.cs:44]: `dictionary?.IsEmpty == false`

File: src/Nethermind/Nethermind.State/PersistentStorageProvider.cs [hunk 3]
```diff
@@ -330,7 +330,6 @@ public void CommitTrees(long blockNumber)
             _toUpdateRoots.Clear();
             // only needed here as there is no control over cached storage size otherwise
             _storages.Clear();
-            _preBlockCache?.NoResizeClear();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.State/PersistentStorageProvider.cs:333]: Not needed now?

File: src/Nethermind/Nethermind.State/StateProvider.cs [hunk 4]
```diff
@@ -857,7 +857,6 @@ public void CommitTree(long blockNumber)
             }
 
             _tree.Commit(blockNumber);
-            _preBlockCache?.NoResizeClear();
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.State/StateProvider.cs:860]: Not needed now?

File: src/Nethermind/Nethermind.Core/Collections/CollectionExtensions.cs (hunk 2 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Core/Collections/CollectionExtensions.cs:44]: > dictionary?.IsEmpty == false

Crashes with that 

`dictionary?.IsEmpty ?? true` works though

**Comment by benaadams** [comment]
Something not happy with this. Will make a simpler PR

**Comment by benaadams** [comment]
Simpler that is kinda and optimiaztion bug fix https://github.com/NethermindEth/nethermind/pull/7442

**Pull request closed without merging**